src/uct/ib/mlx5/rc/rc_mlx5_iface.c

```diff
@@ -928,6 +928,11 @@ UCS_CLASS_INIT_FUNC(uct_rc_mlx5_iface_t,
                     !!(md->flags & UCT_IB_MLX5_MD_FLAG_DP_ORDERING_OOO_RW_RC),
                     !!(md->flags & UCT_IB_MLX5_MD_FLAG_DP_ORDERING_FORCE),
                     &config->rc_mlx5_common.super);
+    if (uct_ib_iface_is_roce(&self->super.super.super) &&
+        (self->super.super.super.config.dp_ordering_ooo == UCS_TERNARY_LAST)) {
+        ucs_error("Can't set RoCE DP ordering mode");
+        return UCS_ERR_INVALID_PARAM;
+    }
```

Comment:
can we share code with dc_mlx5.c? maybe move the error print to uct_rc_mlx5_dp_ordering_ooo_get?

something like
```c
status = uct_rc_mlx5_dp_ordering_init(&self->super, 
                      UCT_IB_MLX5_MD_FLAG_DP_ORDERING_OOO_RW_RC, 
                      &config->rc_mlx5_common);
if (status != UCS_OK) {
    return status;
}
```

Reply:
fixed